### PR TITLE
Added multi_step_lookup.py

### PR DESCRIPTION
**Pull request by dev-bon**
Added multi_step_lookup.py script

- Added output parameters to output either json and csv
- Also implemented adding of drawing of bounding boxes for image results

### CODE REVIEW

**Comment by marcbelmont** [comment]
@dev-bon, it would be good to have this feature directly in https://github.com/parkpow/deep-license-plate-recognition/blob/master/plate_recognition.py
Can you do that next?

File: plate_recognition.py [hunk 1]
```diff
@@ -289,30 +438,8 @@ def main():
...
-            if args.annotate_images:
-                annotated_image.save(
-                    path.with_name(f"{path.stem}_annotated{path.suffix}")
-                )
-
-        results.append(api_res)
-        if args.crop_lp or args.crop_vehicle:
-            save_cropped(api_res, path, args)
+        if Path(path).is_file():
+            results.append(process_image(path, args, engine_config))
```
Comment (marcbelmont) [inline plate_recognition.py:442]: I would rather separate the implementation into `process_full_image` and `process_split_image`. The code for `args.show_boxes` can remain in `main`. This makes the single image process simpler to read and understand.

Comment (dev-bon) [reply plate_recognition.py:442]: Got this, sure will refactor this Sir.

File: plate_recognition.py [hunk 2]
```diff
@@ -289,30 +442,22 @@ def main():
...
-        if (args.show_boxes or args.annotate_images) and "results" in api_res:
+        if Path(path).is_file():
+            if args.split_image:
+                results.append(process_split_image(path, args, engine_config))
+            else:
+                results.append(process_full_image(path, args, engine_config))
+        if args.show_boxes or args.annotate_images and "results" in results:
             image = Image.open(path)
-            annotated_image = draw_bb(image, api_res["results"], None, text_function)
+            annotated_image = draw_bb(image, results["results"], None, text_function)
```
Comment (marcbelmont) [inline plate_recognition.py:452]: `results` is a list with one element per file. I believe we just want the last one.

```suggestion
            annotated_image = draw_bb(image, results["results"][-1], None, text_function)
```

File: plate_recognition.py [hunk 3]
```diff
@@ -289,30 +442,22 @@ def main():
...
-                mmc=args.mmc,
-            )
-
-        if (args.show_boxes or args.annotate_images) and "results" in api_res:
+        if Path(path).is_file():
+            if args.split_image:
+                results.append(process_split_image(path, args, engine_config))
+            else:
+                results.append(process_full_image(path, args, engine_config))
+        if args.show_boxes or args.annotate_images and "results" in results:
```
Comment (marcbelmont) [inline plate_recognition.py:450]:
```suggestion
        if (args.show_boxes or args.annotate_images) and "results" in results[-1]:
```
what do you think?

File: multi_step_lookup.py
Comment (marcbelmont) [inline multi_step_lookup.py:1]: Let's remove this file since the functionality is now in the main script.

**Review by marcbelmont — CHANGES_REQUESTED** [review]

File: plate_recognition.py (hunk 2 above)
Comment (dev-bon) [reply plate_recognition.py:452]: This causes errors Sir, tried to have this but the error is: TypeError: string indices must be integers

based on usage, draw_bb is a function that expects a list

File: plate_recognition.py (hunk 3 above)
Comment (dev-bon) [reply plate_recognition.py:450]: Same goes for this, results should be a list.

File: multi_step_lookup.py
Comment (dev-bon) [reply multi_step_lookup.py:1]: Removed it now sir, I had also placed the annotating in a function, revisited my code and it does not output images, it works now this way.

https://github.com/parkpow/deep-license-plate-recognition/blob/08281c0151398b31c51230a920c63cb8240e42da/plate_recognition.py#L352

File: plate_recognition.py [hunk 4]
```diff
@@ -289,30 +456,11 @@ def main():
...
-
-        results.append(api_res)
-        if args.crop_lp or args.crop_vehicle:
-            save_cropped(api_res, path, args)
+        if Path(path).is_file():
+            if args.split_image:
+                results.append(process_split_image(path, args, engine_config))
+            else:
+                results.append(process_full_image(path, args, engine_config))
     if args.output_file:
```
Comment (marcbelmont) [inline plate_recognition.py:464]: did you check if this parameter works when combined with `--split-image`?

Comment (dev-bon) [reply plate_recognition.py:464]: Yes, works with and without the parameter Sir

**Review by marcbelmont — APPROVED** [review]

**Pull request merged**